lib/helpers.bash

```diff
@@ -283,12 +289,12 @@ _bash-it-update-() {
 
     if [[ $silent ]]; then
       echo "Updating to ${TARGET}($(git log -1 --format=%h "${TARGET}"))..."
-      _bash-it_pull_and_update_inner $TARGET $version
+      _bash-it_pull_and_update_inner $TARGET $version $old_pwd
     else
       read -e -n 1 -p "Would you like to update to ${TARGET}($(git log -1 --format=%h "${TARGET}"))? [Y/n] " RESP
       case $RESP in
         [yY]|"")
-          _bash-it_pull_and_update_inner $TARGET $version
+          _bash-it_pull_and_update_inner $TARGET $version $old_pwd
```

Comment:
So the `bash-it restart` in the called function still allows the function to finish (i.e returns to function after restart), but doesn't remember who called the function (i.e doesn't return here after the function finishes)?

This seems odd to me.  Is there a quick test script that can show this behavior in isolation? ie.:
* Function A calls function B
* Function B does _something_ that still returns control to Function B when done, but doesn't return to function A when function B returns?

Reply:
`bash-it restart` does not return control to the function. It execs into a new bash instance!
Maybe we should document it better that a call to `bash-it restart` does not return

Reply:
OH, I misread -- We `cd` **before** restart ... Yes I get it now.